packages/flutter/lib/src/rendering/object.dart

```diff
@@ -2310,20 +2309,13 @@ abstract class RenderObject with DiagnosticableTreeMixin implements HitTestTarge
       assert(_debugRelayoutBoundaryAlreadyMarkedNeedsLayout());
       return;
     }
-    if (_relayoutBoundary == null) {
-      _needsLayout = true;
+    _needsLayout = true;
+    if (_isRelayoutBoundary != true) {
+      assert(parent != null || _isRelayoutBoundary == null);
       if (parent != null) {
-        // _relayoutBoundary is cleaned by an ancestor in RenderObject.layout.
-        // Conservatively mark everything dirty until it reaches the closest
-        // known relayout boundary.
```

Comment:
Isn't this comment still valid for parts of the branch (i.e. when _isRelayoutBoundary == null)?